node/src/main/scala/coop/rchain/node/node.scala

```diff
@@ -70,36 +69,7 @@ class NodeRuntime(conf: Configuration, host: String, scheduler: Scheduler) {
   }
 
   println(s"Using data_dir: ${dataDirFile.getAbsolutePath}")
-
-  // Generate certificate if not provided as option or in the data dir
-  if (!conf.tls.customCertificateLocation
-      && !conf.tls.certificate.toFile.exists()) {
-    println(s"No certificate found at path ${conf.tls.certificate}")
-    println("Generating a X.509 certificate for the node")
-
-    import coop.rchain.shared.Resources._
-    // If there is a private key, use it for the certificate
-    if (conf.tls.key.toFile.exists()) {
-      println(s"Using secret key ${conf.tls.key}")
-      Try(CertificateHelper.readKeyPair(conf.tls.key.toFile)) match {
-        case Success(keyPair) =>
-          withResource(new java.io.PrintWriter(conf.tls.certificate.toFile)) {
-            _.write(CertificatePrinter.print(CertificateHelper.generate(keyPair)))
-          }
-        case Failure(e) =>
-          println(s"Invalid secret key: ${e.getMessage}")
-      }
-    } else {
-      println("Generating a PEM secret key for the node")
-      val keyPair = CertificateHelper.generateKeyPair(conf.tls.secureRandomNonBlocking)
-      withResource(new java.io.PrintWriter(conf.tls.certificate.toFile)) { pw =>
-        pw.write(CertificatePrinter.print(CertificateHelper.generate(keyPair)))
-      }
-      withResource(new java.io.PrintWriter(conf.tls.key.toFile)) { pw =>
-        pw.write(CertificatePrinter.printPrivateKey(keyPair.getPrivate))
-      }
-    }
-  }
+  transport.generateSertificateIfAbsent(conf.tls)
```

Comment:
Am I reading this correctly that `generateSertificateIfAbsent` here will become generateSertificateIfAbsent[Tls]` which means it will require `Iso[Tls, Tls]` ?

Reply:
Nope. It will require `Iso[coop.rchain.node.configuration.Tls, coop.rchain.comm.transport.Tls]`.